### PR TITLE
Feature/space avatar in activities

### PR DESCRIPTION
**Pull request by aboughzela**

### CODE REVIEW

File: extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl [hunk 1]
```diff
@@ -162,6 +162,23 @@
   }
   def currentTime = System.currentTimeMillis();
   String viewActivityTip = _ctx.appRes("UIActivity.msg.ViewActivity");
+
+  def spaceSourceURI, spaceName, spaceImageSource;
```
Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl]: Can we move this code in another location to avoid adding to much logic in templates and to avoid duplicating it in each activity template ?

Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl]: I see this change only in UIDefaultActivity.gtmpl and UIRelationshipActivity.gtmpl. Because of that, on wiki activity types for example, we do not have the space avatar in the timeline. It should be in all activity types templates.

File: webapp/resources/src/main/webapp/javascript/eXo/UISpaceInfoPopup.js [hunk 2]
```diff
@@ -0,0 +1,435 @@
+/*
+ * TipTip
```
Comment (thomasdelhomenie) [inline webapp/resources/src/main/webapp/javascript/eXo/UISpaceInfoPopup.js]: The header does not seem to be related to the code.

File: webapp/resources/src/main/webapp/javascript/eXo/UIUserProfilePopup.js [hunk 3]
```diff
@@ -337,6 +337,7 @@
                     tdAvatar.append(aAvatar.append(img));
 
                     var tdProfile = $("<td/>");
+                    tdProfile.css({'overflow': 'hidden','white-space': 'nowrap','text-overflow': 'ellipsis', 'max-width': '20px'});
```
Comment (thomasdelhomenie) [inline webapp/resources/src/main/webapp/javascript/eXo/UIUserProfilePopup.js]: Can't we move style in css files (in platform-ui) instead ?

File: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIProfile.js [hunk 4]
```diff
@@ -48,13 +47,44 @@ var UIProfile = {
...
+    $.each(UIProfile.labels, function(key) {
+      UIProfile.labels[key] =  window.decodeURIComponent(UIProfile.labels[key]);
+    });
+
+    // User Profile Popup initialize
+    var portal = eXo.social.portal;
+    var spaceRestUrl = '//' + window.location.host + portal.context + '/' + portal.rest + portal.context + '/social/spaces' + '/getSpaceInfo/{0}.json';
+
+    var container = $('#' + uicomponentId).closest('.PORTLET-FRAGMENT');
+    var spaceLinks = $(container).find('a:[href*="/g/:spaces:"]');
```
Comment (thomasdelhomenie) [inline webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIProfile.js]: Looks fragile. What if the activity contains a link to an URL in a space (for example http://..../portal/g/:spaces:myspace/wiki/MyPage1). We should instead display the space popup only on the space avatar element, like we do with user popup in chat for example : https://github.com/exo-addons/chat-application/blob/develop/application/src/main/webapp/js/chat.js#L3190

File: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIProfile.js [hunk 5]
```diff
@@ -48,13 +47,44 @@ var UIProfile = {
...
   },
+  initSpaceInfoPopup : function(uicomponentId, labels) {
+    //
+    UIProfile.labels = $.extend(true, {}, UIProfile.labels, labels);
+    $.each(UIProfile.labels, function(key) {
+      UIProfile.labels[key] =  window.decodeURIComponent(UIProfile.labels[key]);
+    });
+
+    // User Profile Popup initialize
+    var portal = eXo.social.portal;
```
Comment (thomasdelhomenie) [inline webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIProfile.js]: We should use eXo.env.portal instead IMO.

File: component/webui/src/main/resources/locale/social/Webui_en.properties [hunk 6]
```diff
@@ -20,6 +20,9 @@
 #####################################################################################
 UIActivitiesLoader.label.Not_Any_Updates_Posted_Yet=No updates have been posted yet.
 UIActivitiesLoader.label.Load_More=Load More
+UIActivitiesLoader.label.Space_INFO_Followers=Followers
+UIActivitiesLoader.label.Space_INFO_Leave_Space=Leave Space
```
Comment (thomasdelhomenie) [inline component/webui/src/main/resources/locale/social/Webui_en.properties]: I think we already have a label for "Leaving a space", we should re-use it to be consistent IMO.

File: component/webui/src/main/resources/locale/social/Webui_en.properties [hunk 7]
```diff
@@ -20,6 +20,9 @@
 #####################################################################################
 UIActivitiesLoader.label.Not_Any_Updates_Posted_Yet=No updates have been posted yet.
 UIActivitiesLoader.label.Load_More=Load More
+UIActivitiesLoader.label.Space_INFO_Followers=Followers
```
Comment (thomasdelhomenie) [inline component/webui/src/main/resources/locale/social/Webui_en.properties]: I think we should use the term "Members" instead of "Followers" here, to be consistent with the platform. To be checked with Hamadi.

File: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIProfile.js (hunk 5 above)
Comment (aboughzela) [inline webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIProfile.js]: I think eXo.social.portal is more generic 
see variable definition
  if (eXo.env) {
    var portal = eXo.env.portal
    
    eXo.social.portal = {
      rest : (portal.rest) ? portal.rest : 'rest',
      portalName : (portal.portalName) ? portal.portalName : 'intranet',
      context : (portal.context) ? portal.context : '/portal',
      accessMode : (portal.accessMode) ? portal.accessMode : 'public',
      userName : (portal.userName) ? portal.userName : ''
    };
  }
WYT ?

File: component/webui/src/main/resources/locale/social/Webui_en.properties (hunk 7 above)
Comment (aboughzela) [inline component/webui/src/main/resources/locale/social/Webui_en.properties]: >> To be checked with Hamadi.
It's OK to change it (confirmed by Hamadi)

File: component/webui/src/main/resources/locale/social/Webui_en.properties (hunk 6 above)
Comment (aboughzela) [inline component/webui/src/main/resources/locale/social/Webui_en.properties]: I used the same label defined on spec and design  mockup  (It's OK to change it (confirmed by Hamadi))

File: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIProfile.js (hunk 5 above)
Comment (thomasdelhomenie) [inline webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIProfile.js]: Actually I don't see why we have both. eXo.env.portal should be used everywhere where we need the context path or the rest context path IMO.

File: component/webui/src/main/resources/locale/social/Webui_en.properties (hunk 7 above)
Comment (thomasdelhomenie) [inline component/webui/src/main/resources/locale/social/Webui_en.properties]: You mean Hamadi is ok to not be consistent (so to use Followers instead of Members) ?

Comment (aboughzela) [inline component/webui/src/main/resources/locale/social/Webui_en.properties]: Hamadi is ok to use Members

File: component/webui/src/main/resources/locale/social/Webui_en.properties (hunk 6 above)
Comment (thomasdelhomenie) [inline component/webui/src/main/resources/locale/social/Webui_en.properties]: ok

File: extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl (hunk 1 above)
Comment (aboughzela) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl]: this change is existed (added on space avatar version 1 ) in all template related to space activity (wiki, post ..) see integration PR

Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl]: I don't see this piece of code in develop (space avatar v1 has been merged to develop). Can you point me to an example please ?

Comment (aboughzela) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl]: https://github.com/exodev/integration/commit/5cd39c00a64ef96cc81f5f3dd7e0bfeb5b13cd41

Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl]: Actually I was talking about the piece of code below. This piece of code does not exist in develop, it has been added in this branch.

Comment (aboughzela) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl]: no it also exist in develop integration https://github.com/exodev/integration/blob/develop/integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/plugin/space/FileUIActivity.gtmpl#L278

Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UIDefaultActivity.gtmpl]: ok, right.
But I still think it should be put in a common location. Is it possible ?

File: component/webui/src/main/resources/locale/social/Webui_en.properties (hunk 6 above)
Comment (aboughzela) [inline component/webui/src/main/resources/locale/social/Webui_en.properties]: PR updated please review
